app/dashboard/models.py

```diff
@@ -1142,129 +1144,153 @@ def graph_capacities(simulations, y_variables):
             if qs_asset_results.count() == 1:
                 el.update(qs_asset_results.values("optimized_capacity", "total_flow", "direction").get())
             elif qs_asset_results.count() > 1:
-                qs_asset_results = qs_asset_results.filter(optimized_capacity__isnull=False)
-                if qs_asset_results.count() == 1:
-                    el.update(qs_asset_results.values("optimized_capacity", "total_flow", "direction").get())
-                elif qs_asset_results.count() > 1:
-                    raise ValueError("should not have too much labels")
-            records.append(el)
-
-        wacc = simulation.scenario.project.economic_data.discount
-        df = pd.DataFrame.from_records(records)
-
-        # assign optimized capacity to storage components
-        storages = df.parent_asset__name.dropna().unique()
-        for ess in storages:
-            df.loc[
-                (df.parent_asset__name == ess) & (df.direction.isna() == True),
-                "optimized_capacity",
-            ] = df.loc[
-                (df.parent_asset__name == ess) & (df.direction == "out"),
-                "optimized_capacity",
-            ].values[0]
-
-        df = df.fillna(0)
-        # TODO costs for batteries are skewed as battery capacity does not exists in fancy results
-        # TODO costs for dso not implemented yet
-        df["capex_total"] = df.apply(
-            lambda x: (x.installed_capacity + x.optimized_capacity)
-            * x.capex_var
-            * (wacc * (1 + wacc) ** x.lifetime)
-            / ((1 + wacc) ** x.lifetime - 1),
-            axis=1,
-        )
-        df["opex_fix_total"] = df.apply(lambda x: (x.installed_capacity + x.optimized_capacity) * x.opex_fix, axis=1)
-        df["opex_var_total"] = df.apply(lambda x: x.total_flow * x.opex_var, axis=1)
+                raise ValueError("should not have too much labels")
+        records.append(el)
+
+    wacc = simulation.scenario.project.economic_data.discount
+    df = pd.DataFrame.from_records(records)
+
+    # assign optimized capacity to storage components
+    storages = df.parent_asset__name.dropna().unique()
+    for ess in storages:
+        df.loc[
+            (df.parent_asset__name == ess) & (df.direction.isna() == True),
+            "optimized_capacity",
+        ] = df.loc[
+            (df.parent_asset__name == ess) & (df.direction == "out"),
+            "optimized_capacity",
+        ].values[0]
+
+    df = df.fillna(0)
+    # TODO costs for batteries are skewed as battery capacity does not exists in fancy results
+    # TODO costs for dso not implemented yet
+    # why is it this formula and why is the installed capacity included in capex, i dont understand
+    df["capex_total"] = df.apply(
+        lambda x: (x.installed_capacity + x.optimized_capacity)
+        * x.capex_var
+        * (wacc * (1 + wacc) ** x.lifetime)
+        / ((1 + wacc) ** x.lifetime - 1),
+        axis=1,
+    )
```

Comment:
i decided to calculate the capex manually again with optimized capacity because i dont really understand what is happening here. this function returns a value much lower than capex/unit*optimized_capacity. It is not super relevant because we are not going to use the costs graph but it also makes the graph not match the manually calculated values.